clarity/src/vm/analysis/analysis_db.rs

```diff
@@ -185,14 +221,18 @@ impl<'a> AnalysisDatabase<'a> {
         &mut self,
         contract_identifier: &QualifiedContractIdentifier,
         map_name: &str,
+        epoch: &StacksEpochId,
     ) -> CheckResult<(TypeSignature, TypeSignature)> {
         let contract = self
-            .load_contract(contract_identifier)
+            .load_contract_non_canonical(contract_identifier)
             .ok_or(CheckErrors::NoSuchContract(contract_identifier.to_string()))?;
         let map_type = contract
             .get_map_type(map_name)
             .ok_or(CheckErrors::NoSuchMap(map_name.to_string()))?;
-        Ok(map_type.clone())
+        Ok((
+            map_type.0.canonicalize(epoch),
+            map_type.1.canonicalize(epoch),
```

Comment:
Should we also be canonicalizing data-var types?

Reply:
My assumption was that we only need to canonicalize the outward-facing types, which would include public and read-only functions, and defined traits. I'm not actually sure where/how this `get_map_type` method would get used.

Reply:
It looks like this function is not used anywhere - perhaps consider deleting

Reply:
Sounds good to me. Any one have a reason to keep it?

Reply:
Nope -- I think this was vestigial from when Clarity had a function to fetch data from an arbitrary contract's data map. Delete away!

Reply:
Awesome. Deleted in b1343ea55.